src/core/InstrumentFunctions.cpp

```diff
@@ -474,37 +474,19 @@ void InstrumentFunctionArpeggio::processNote( NotePlayHandle * _n )
 		{
 			cur_arp_idx = (cur_frame / arp_frames) % (range + sortOffset);
 		}
-		else if( dir == ArpDirection::Down )
-		{
-			cur_arp_idx = range + sortOffset - (cur_frame / arp_frames) %
-								(range + sortOffset) - 1;
-		}
-		else if (dir == ArpDirection::UpAndDown && (range + sortOffset) > 1)
+		else if ((dir == ArpDirection::UpAndDown || dir == ArpDirection::DownAndUp) && (range + sortOffset) > 1)
 		{
 			// imagine, we had to play the arp once up and then
 			// once down -> makes 2 * range possible notes...
 			// because we don't play the lower and upper notes
 			// twice, we have to subtract 2
-			cur_arp_idx = (cur_frame / arp_frames) % ((range + sortOffset) * 2 - 2);
-			// if greater than range, we have to play down...
-			// looks like the code for arp_dir==DOWN... :)
-			if (cur_arp_idx >= range + sortOffset)
-			{
-				cur_arp_idx = range + sortOffset - cur_arp_idx % (range + sortOffset - 1) - 1;
-			}
-		}
-		else if (dir == ArpDirection::DownAndUp && (range + sortOffset) > 1)
-		{
-			// copied from ArpDirection::UpAndDown above
-			cur_arp_idx = (cur_frame / arp_frames) % ((range + sortOffset) * 2 - 2);
+			cur_arp_idx = (cur_frame / arp_frames) % ((range + sortOffset) * 2 - (2 * static_cast<int>(m_arpRepeatsModel.value())));
 			// if greater than range, we have to play down...
 			// looks like the code for arp_dir==DOWN... :)
```

Comment:
// looks like the code for arp_dir==DOWN... :)

Which was removed so this line should be removed.

Reply:
I will fix this in a few days. Sorry for the waiting.